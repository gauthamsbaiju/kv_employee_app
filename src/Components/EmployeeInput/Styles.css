.input-wrapper1{
    width: calc(100% / 3.5);
    
}

.label{
    margin-top: 0px;
}

.text-input{
    height: 45px;
    width: 100%;
    background: #FAFAFA 0% 0% no-repeat padding-box;
    border: 1px solid #D9D9D9;
    opacity: 1;
    padding-left: 10px;
    border-radius: 5px;
}