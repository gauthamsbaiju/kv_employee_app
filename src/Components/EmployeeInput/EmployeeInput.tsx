import React from 'react';
import './Styles.css';
import { Department } from '../../types';
// import { useNavigate } from 'react-router-dom';
// import Button from '../Button/Button';

export type EmpInputPropsType = {
  onChange: (e, n) => void;
  lable: string;
  placeholder?: string;
  type: 'text' | 'dropdown' | 'address' | 'id' | 'password';
  options: string[] | Department[] | null;
  add?: {
    address_line_1: string;
    address_line_2: string;
    city: string;
    state: string;
    country: string;
    pincode: string;
  };
  defaultValue?: string;
};

const EmployeeInput: React.FC<EmpInputPropsType> = (props) => {
  console.log('def val: ');
  // console.log(props.defaultValue);
  // const navigate = useNavigate();
  // const create = (event) => {
  //   console.log(event);
  //   console.log('Submitted');
  //   navigate('/employee');
  // };

  // const cancel = (event) => {
  //   console.log(event);
  //   console.log('Submitted');
  //   navigate('/employee');
  // };
  let addr: string[] = [];

  if (props.type === 'address' && props.defaultValue) {
    addr = props.defaultValue.split('_');
    console.log('Def val new');
    console.log(addr);
  }

  return (
    <div className='input-wrapper1' data-testid='EmpDetails-test'>
      <label className='label1' data-testid='EmpDetailslabel-test'>
        {props.lable}
      </label>

      {props.type === 'text' && (
        <input
          type={props.type}
          placeholder={props.placeholder}
          // value={props.defaultValue}
          className='text-input'
          onChange={(e) => props.onChange(e, 0)}
          data-testid='EmpDetailsProps-test'
        ></input>
      )}

      {props.type === 'password' && (
        <input
          type={props.type}
          placeholder={props.placeholder}
          // value={props.defaultValue}
          className='text-input'
          onChange={(e) => props.onChange(e, 0)}
        ></input>
      )}

      {props.type === 'id' && (
        <input
          disabled
          type={props.type}
          placeholder={props.placeholder}
          // value={props.defaultValue}
          className='text-input'
          onChange={(e) => props.onChange(e, 0)}
        ></input>
      )}

      {props.type === 'dropdown' && props.lable === 'Department' && (
        <select
          placeholder={props.placeholder}
          // value={props.defaultValue}
          className='text-input'
          onChange={(e) => props.onChange(e, 0)}
        >
          <option value='' disabled selected>
            {props.placeholder}
          </option>
          {props.options.map((item, index) => {
            return <option key={index}>{item.name}</option>;
          })}
        </select>
      )}
      {props.type === 'dropdown' && props.lable !== 'Department' && (
        <select
          placeholder={props.placeholder}
          // value={props.defaultValue}
          className='text-input'
          onChange={(e) => props.onChange(e, 0)}
        >
          <option value='' disabled selected>
            {props.placeholder}
          </option>
          {props.options.map((item, index) => {
            return <option key={index}>{item}</option>;
          })}
        </select>
      )}

      {props.type === 'address' && (
        <div className='address-wrapper'>
          <input
            type={props.type}
            placeholder={props.add.address_line_1}
            // value={addr[1]}
            className='address-input'
            onChange={(e) => props.onChange(e, 1)}
          ></input>
          <input
            type={props.type}
            placeholder={props.add.address_line_2}
            // value={addr[2]}
            className='address-input'
            onChange={(e) => props.onChange(e, 2)}
          ></input>
          <input
            type={props.type}
            placeholder={props.add.city}
            // value={addr[0]}
            className='address-input'
            onChange={(e) => props.onChange(e, 3)}
          ></input>
          <input
            type={props.type}
            placeholder={props.add.state}
            // value={addr[0]}
            className='address-input'
            onChange={(e) => props.onChange(e, 4)}
          ></input>
          <input
            type={props.type}
            placeholder={props.add.country}
            // value={addr[0]}
            className='address-input'
            onChange={(e) => props.onChange(e, 5)}
          ></input>
          <input
            type={props.type}
            placeholder={props.add.pincode}
            // value={addr[0]}
            className='address-input'
            onChange={(e) => props.onChange(e, 6)}
          ></input>
          {/* <div className='btndiv'>
            <Button label='Create' className='createbtn' onClick={create} type='submit'></Button>
            <Button label='Cancel' className='cancelbtn' onClick={cancel} type='submit'></Button>
          </div> */}
        </div>
      )}
    </div>
  );
};

export default EmployeeInput;
