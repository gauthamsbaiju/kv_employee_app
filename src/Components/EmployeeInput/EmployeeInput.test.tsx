import { fireEvent, render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import EmployeeInput, { EmpInputPropsType } from './EmployeeInput';

describe('Test for employee input component', () => {
  test('Test for employee input label rendered', () => {
    const onChange = jest.fn();
    const EmpInputProps: EmpInputPropsType = {
      lable: ' Emp Input',
      placeholder: 'Emp Input',
      type: 'text',
      options: ['active', 'inactive'],
      onChange
    };

    render(<EmployeeInput {...EmpInputProps} />);
    const element = screen.getByTestId('EmpDetailslabel-test');

    if (element) expect(element).toHaveTextContent('Emp Input');
  });

  test('Test for snapshot rendered', () => {
    const onChange = jest.fn();
    const EmpInputProps: EmpInputPropsType = {
      lable: ' Emp Input',
      placeholder: 'Emp Input',
      type: 'text',
      options: ['active', 'inactive'],
      onChange
    };

    render(<EmployeeInput {...EmpInputProps} />);
    const element = screen.getByTestId('EmpDetails-test');

    expect(element).toMatchSnapshot();
  });

  test('Test for employee input type', () => {
    const onChange = jest.fn();
    const EmpInputProps: EmpInputPropsType = {
      lable: ' Emp Input',
      placeholder: 'Emp Input',
      type: 'text',
      options: ['active', 'inactive'],
      onChange
    };

    render(<EmployeeInput {...EmpInputProps} />);
    const element = screen.getByTestId('EmpDetailsProps-test');

    expect(element).toHaveAttribute('type', 'text');
  });

  test('Test for employee input placeholder', () => {
    const onChange = jest.fn();
    const EmpInputProps: EmpInputPropsType = {
      lable: ' Emp Input',
      placeholder: 'Emp Input',
      type: 'text',
      options: ['active', 'inactive'],
      onChange
    };

    render(<EmployeeInput {...EmpInputProps} />);
    const element = screen.getByTestId('EmpDetailsProps-test');

    expect(element).toHaveAttribute('placeholder', 'Emp Input');
  });

  //   test('Test for employee input options', () => {
  //     const onChange = jest.fn();
  //     const EmpInputProps: EmpInputPropsType = {
  //       lable: ' Emp Input',
  //       placeholder: 'Emp Input',
  //       type: 'text',
  //       options: ['active', 'inactive'],
  //       onChange
  //     };

  //     render(<EmployeeInput {...EmpInputProps} />);
  //     const element = screen.getByTestId('EmpDetailsProps-test');

  //     expect(element).toHaveAttribute('options', ['active', 'inactive']);
  //   });

  test('Test for on change', () => {
    const onChange = jest.fn();
    const EmpInputProps: EmpInputPropsType = {
      lable: ' Emp Input',
      placeholder: 'Emp Input',
      type: 'text',
      options: ['active', 'inactive'],
      onChange
    };

    render(<EmployeeInput {...EmpInputProps} />);
    const element = screen.getByTestId('EmpDetailsProps-test');

    fireEvent.change(element);
    expect(onChange).toBeCalled;
  });
});
