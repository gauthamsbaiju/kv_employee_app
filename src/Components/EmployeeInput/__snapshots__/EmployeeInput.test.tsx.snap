// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for employee input component Test for snapshot rendered 1`] = `
<div
  class="input-wrapper1"
  data-testid="EmpDetails-test"
>
  <label
    class="label1"
    data-testid="EmpDetailslabel-test"
  >
     Emp Input
  </label>
  <input
    class="text-input"
    data-testid="EmpDetailsProps-test"
    placeholder="Emp Input"
    type="text"
  />
</div>
`;
